uint16 gpioId
---
int32 mode
bool hasSucceeded